pkg/util/pod/pod.go

```diff
@@ -65,35 +65,42 @@ func gateIndex(p *corev1.Pod, gateName string) int {
 }
 
 var (
-	errLabelNotFound = errors.New("label not found")
-	errInvalidUInt   = errors.New("invalid unsigned integer")
-	errValidation    = errors.New("validation error")
+	ErrLabelNotFound = errors.New("label not found")
+	ErrInvalidUInt   = errors.New("invalid unsigned integer")
+	ErrValidation    = errors.New("validation error")
 )
 
+func IgnoreLabelNotFoundError(err error) error {
+	if errors.Is(err, ErrLabelNotFound) {
+		return nil
+	}
+	return err
+}
+
 func ReadUIntFromLabel(obj client.Object, labelKey string) (*int, error) {
-	return ReadUIntFromLabelBelowBound(obj, labelKey, math.MaxInt)
+	return ReadUIntFromLabelWithMax(obj, labelKey, math.MaxInt)
 }
 
-func ReadUIntFromLabelBelowBound(obj client.Object, labelKey string, bound int) (*int, error) {
+func ReadUIntFromLabelWithMax(obj client.Object, labelKey string, max int) (*int, error) {
 	value, found := obj.GetLabels()[labelKey]
 	kind := obj.GetObjectKind().GroupVersionKind().Kind
 	if !found {
-		return nil, fmt.Errorf("%w: no label %q for %s %q", errLabelNotFound, labelKey, kind, klog.KObj(obj))
+		return nil, fmt.Errorf("%w: no label %q for %s %q", ErrLabelNotFound, labelKey, kind, klog.KObj(obj))
 	}
-	intValue, err := readUIntFromStringBelowBound(value, bound)
+	intValue, err := readUIntFromStringWithMax(value, max)
 	if err != nil {
 		return nil, fmt.Errorf("incorrect label value %q for %s %q: %w", value, kind, klog.KObj(obj), err)
 	}
 	return intValue, nil
 }
 
-func readUIntFromStringBelowBound(value string, bound int) (*int, error) {
+func readUIntFromStringWithMax(value string, max int) (*int, error) {
 	uintValue, err := strconv.ParseUint(value, 10, 0)
 	if err != nil {
-		return nil, fmt.Errorf("%w: %s", errInvalidUInt, err.Error())
+		return nil, fmt.Errorf("%w: %s", ErrInvalidUInt, err.Error())
 	}
-	if uintValue > uint64(bound) {
-		return nil, fmt.Errorf("%w: value should be less than %d", errValidation, bound)
+	if uintValue >= uint64(max) {
```

Comment:
Why do we want to return error if `uintValue == max`?

Reply:
+1, also why is the PR changing the function, I think for pod-group validation we can reuse `ReadUIntFromLabelBelowBound` without adding the +1

Reply:
> Why do we want to return error if `uintValue == max`?

On previous realization we are checking for equal 

https://github.com/kubernetes-sigs/kueue/pull/3717/files#diff-dfb49586a8522fa91d733051fd3b7e4b3ff174907898cf249ab16e2620976a5dL572-L574

Reply:
> +1, also why is the PR changing the function, I think for pod-group validation we can reuse `ReadUIntFromLabelBelowBound` without adding the +1

In that case on pod controller we need to set -1.

Reply:
The previous implementation was pod specific and this is more generic function. Agree, that in pod controller we should call this function with `GroupCount - 1`

Reply:
Not sure that it is correct way. Imagine situation that you have podGroupTotalCount=1. Minus 1 it would be 0. And we can't compare with -1.

Reply:
I don't get it, pod group pod index can be only from 0 to group size -1, so using `ReadUIntFromLabelBelowBound` should be ok as is.

Reply:
Done